puppet-blacksmith.gemspec

```diff
@@ -18,9 +18,10 @@ Gem::Specification.new do |s|
   s.required_ruby_version = '>= 2.4.0'
 
   s.add_runtime_dependency 'rest-client', '~>2.0'
+  s.add_development_dependency 'bundler'
   s.add_development_dependency 'rake'
-  s.add_development_dependency 'pdk'
-  s.add_development_dependency 'puppet', '>=2.7.16'
+  s.add_development_dependency 'pdk', '~>1.17.0'
```

Comment:
Why does it need to be pinned to 1.17.x?

Reply:
This was the root cause of the specs failing. Unless you forced the dependencies to resolve to the latest pdk. The puppet_spec_helper gem was unable to find pdk. You can see an example in this build https://travis-ci.com/jmangt/puppet-blacksmith/jobs/295415528 
```   
+rake aborted!

      +Errno::ENOENT: No such file or directory - pdk

      +/home/travis/build/jmangt/puppet-blacksmith/vendor/bundle/ruby/2.6.0/gems/puppetlabs_spec_helper-2.14.1/lib/puppetlabs_spec_helper/rake_tasks.rb:150:in ``'

      +/home/travis/build/jmangt/puppet-blacksmith/vendor/bundle/ruby/2.6.0/gems/puppetlabs_spec_helper-2.14.1/lib/puppetlabs_spec_helper/rake_tasks.rb:150:in `rescue in block (2 levels) in <top (required)>'

      +/home/travis/build/jmangt/puppet-blacksmith/vendor/bundle/ruby/2.6.0/gems/puppetlabs_spec_helper-2.14.1/lib/puppetlabs_spec_helper/rake_tasks.rb:144:in `block (2 levels) in <top (required)>'

      +/home/travis/build/jmangt/puppet-blacksmith/vendor/bundle/ruby/2.6.0/gems/puppetlabs_spec_helper-2.14.1/lib/puppetlabs_spec_helper/rake_tasks.rb:129:in `block in <top (required)>'

      +/home/travis/build/jmangt/puppet-blacksmith/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/exe/rake:27:in `<top (required)>'

      +/home/travis/build/jmangt/puppet-blacksmith/vendor/bundle/ruby/2.6.0/bin/ruby_executable_hooks:24:in `eval'

      +/home/travis/build/jmangt/puppet-blacksmith/vendor/bundle/ruby/2.6.0/bin/ruby_executable_hooks:24:in `<main>'

      +

      +Caused by:

      +LoadError: cannot load such file -- pdk/module/build
```

Reply:
Interesting, but I still think it can be pinned to `~> 1.17` so we're not stuck on the 1.17.x line when 1.18 comes out.